cirq-core/cirq/transformers/transformer_api.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
Optional:  Should this file be called api.py to avoid smurf naming?

Reply:
I think it's better to call it `transformer_api` because searching file named `api` in IDEs during development would be tricky. We are also using similar naming elsewhere - like `transformer_primitives` instead of just `primitives`.